From 193d31018790f8c0cfd83ff40268dc1f8612017e Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 1 Aug 2019 09:20:05 +0000
Subject: [PATCH] snapshot_toolchain.gni: use host_toolchain when building
 mksnapshot

* otherwise it's linked against the glibc from target sysroot, but when executed during the build
  it fails when host has older glibc as shown in:
  http://caprica.lgsvl.com:8080/Errors/Details/1555329
---
 snapshot_toolchain.gni | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/snapshot_toolchain.gni b/snapshot_toolchain.gni
index f1c6909439..a6fc1235a1 100644
--- a/snapshot_toolchain.gni
+++ b/snapshot_toolchain.gni
@@ -53,13 +53,13 @@ if (v8_snapshot_toolchain == "") {
   if (current_os == host_os && current_cpu == host_cpu) {
     # This is not a cross-compile, so build the snapshot with the current
     # toolchain.
-    v8_snapshot_toolchain = current_toolchain
+    v8_snapshot_toolchain = host_toolchain
   } else if (current_os == host_os && current_cpu == "x86" &&
              host_cpu == "x64") {
     # This is an x64 -> x86 cross-compile, but x64 hosts can usually run x86
     # binaries built for the same OS, so build the snapshot with the current
     # toolchain here, too.
-    v8_snapshot_toolchain = current_toolchain
+    v8_snapshot_toolchain = host_toolchain
   } else if (current_os == "win" && host_os == "mac" && is_clang) {
     # This is a mac -> win cross-compile, which is only supported w/ clang.
     v8_snapshot_toolchain = "//build/toolchain/mac:clang_${v8_current_cpu}"
